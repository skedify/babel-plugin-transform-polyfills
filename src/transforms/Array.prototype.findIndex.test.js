describe('Array.prototype.findIndex', () => {
  it('should replace Array.prototype.findIndex with a polyfill', () => {
    const result = compile(
      `
      const list = [1,2,3,4,5]
      if (list.findIndex(n => n === 3) === 2) {
        console.log('success')
      } else {
        console.log('failure')
      }
    `
    )

    return expect(result).toMatchSnapshot()
  })

  it('should not import polyfill more than once', () => {
    const result = compile(
      `
      const list = [1,2,3,4,5]
      if (list.findIndex(n => n === 3) === 2) {
        console.log(list.findIndex(3))
      } else {
        console.log('failure')
      }
    `
    )

    return expect(result).toMatchSnapshot()
  })

  it('should work when part of a call chain', () => {
    const result = compile(
      `
      const list = [1,2,3,4,5]
      if (list.slice(2).findIndex(n => n === 3) === 0) {
        console.log('success')
      } else {
        console.log('failure')
      }
    `
    )

    return expect(result).toMatchSnapshot()
  })
})
