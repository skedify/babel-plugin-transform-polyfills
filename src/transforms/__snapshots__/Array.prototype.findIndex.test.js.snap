// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Array.prototype.findIndex should not import polyfill more than once 1`] = `
"import _Array_prototype_findIndex from 'babel-plugin-transform-polyfills/polyfills/Array/prototype/findIndex';

const list = [1, 2, 3, 4, 5];
if (_Array_prototype_findIndex(list, n => n === 3) === 2) {
  console.log(_Array_prototype_findIndex(list, 3));
} else {
  console.log('failure');
}"
`;

exports[`Array.prototype.findIndex should replace Array.prototype.findIndex with a polyfill 1`] = `
"import _Array_prototype_findIndex from 'babel-plugin-transform-polyfills/polyfills/Array/prototype/findIndex';

const list = [1, 2, 3, 4, 5];
if (_Array_prototype_findIndex(list, n => n === 3) === 2) {
  console.log('success');
} else {
  console.log('failure');
}"
`;

exports[`Array.prototype.findIndex should work when part of a call chain 1`] = `
"import _Array_prototype_findIndex from 'babel-plugin-transform-polyfills/polyfills/Array/prototype/findIndex';

const list = [1, 2, 3, 4, 5];
if (_Array_prototype_findIndex(list.slice(2), n => n === 3) === 0) {
  console.log('success');
} else {
  console.log('failure');
}"
`;
